### PR TITLE
Fixed some deprecated call, symfonified the code

### PR DESCRIPTION
**Pull request by ABM-Dan**

### CODE REVIEW

**Comment by Mondane** [comment]
@adamquaile is this MR ready for inclusion?

**Comment by adamquaile** [comment]
@ABM-Dan Apologies for the delay in merging this, and thanks a lot for the update!

File: Form/FieldsetType.php [hunk 1]
```diff
@@ -8,49 +8,47 @@
...
-        ));
+        $resolver
+            ->setDefaults([
+                'legend' => '',
+                'inherit_data' => true,
+                'options' => array(),
+                'fields' => array(),
+                'label' => false,
+            ])
+            ->addAllowedTypes('fields', ['array', 'callable']);
```
Comment (ABM-Dan) [inline Form/FieldsetType.php]: I accidentally used the short array notation, you might want to revert this change.

Comment (adamquaile) [inline Form/FieldsetType.php]: This has support from 5.4 onwards, 5.3 and 5.4 have both been end-of-life for some time, so I'm not too worried about this on this project.

Comment (ABM-Dan) [inline Form/FieldsetType.php]: It's mostly a matter of consistency, either go short syntax everywhere, or long syntax everywhere, mixing feels sloppy, and I'm at fault here. :P

Comment (adamquaile) [inline Form/FieldsetType.php]: Fair enough, then I'd probably go the other way and have short-syntax everywhere, which we can do separately.

**Comment by Mondane** [comment]
With this MR in master,  does it mean a Symfony 3.x compatible release is imminent?
